src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -626,6 +636,161 @@ class MessageHelperTest : public ::testing::Test {
+              (*result)[strings::params][strings::function_id].asInt());
+    EXPECT_EQ(kCorrelationId,
+              (*result)[strings::params][strings::correlation_id].asInt());
+    if (msg.empty()) {
+      EXPECT_FALSE((*result)[strings::msg_params].keyExists("messages"));
+    } else {
+      smart_objects::SmartObject& user_friendly_messages =
+          (*result)[strings::msg_params]["messages"];
+      EXPECT_TRUE(user_friendly_messages.length() == msg.size());
+      std::vector<policy::UserFriendlyMessage>::const_iterator it = msg.begin();
+      for (int i = 0; it != msg.end(); ++i, ++it) {
+        EXPECT_TRUE(it->message_code ==
+                    user_friendly_messages[i]["messageCode"].asString());
+      }
+    }
+  }
+
+  /**
+   * @brief Check LaunchApp message which is sent to HMI
+   * @param urlSchema parameter for filling of message structure .
+   * @param packageName parameter for filling of message structure .
+   */
+  void CheckParametersLaunchAppMessage(const std::string& urlSchema,
+                                       const std::string& packageName) {
+    smart_objects::SmartObjectSPtr result;
+    EXPECT_CALL(mock_application_manager_, ManageMobileCommand(_, _))
+        .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+    MessageHelper::SendLaunchApp(
+        kAppId, urlSchema, packageName, mock_application_manager_);
+
+    ASSERT_TRUE(result.valid());
+    if (!urlSchema.empty() || !packageName.empty()) {
+      EXPECT_TRUE((*result)[strings::msg_params].keyExists(strings::url));
+    } else {
+      EXPECT_FALSE((*result)[strings::msg_params].keyExists(strings::url));
+    }
+
+    if (!urlSchema.empty()) {
+      EXPECT_EQ((*result)[strings::msg_params][strings::url].asString(),
+                urlSchema);
+    } else if (!packageName.empty()) {
+      EXPECT_EQ((*result)[strings::msg_params][strings::url].asString(),
+                packageName);
+    }
+    EXPECT_TRUE(
+        (*result)[strings::msg_params].keyExists(strings::request_type));
+    EXPECT_EQ((*result)[strings::msg_params][strings::request_type].asInt(),
+              mobile_apis::RequestType::LAUNCH_APP);
+    CheckParamsOfMessage(result, true, false);
+    EXPECT_TRUE((*result)[strings::params].keyExists(strings::function_id));
+    EXPECT_EQ((*result)[strings::params][strings::function_id].asInt(),
+              mobile_apis::FunctionID::OnSystemRequestID);
+  }
+
+  /**
+   * @brief Check NaviStartStream message which is sent to HMI
+   * @param url parameter for filling of message structure.
+   * @param video_server_type type of servise that is used for streaming.
+   * @param server_path contains name of pipe, file or adres of socket
+   * @param port contains port for socket.
+   */
+  void CheckParametersNaviStartStreamMessage(
+      const std::string url,
+      const std::string video_server_type,
+      const std::string server_path,
+      const uint16_t port = 0) {
```

Comment:
@dev-gh I do not see the point because I send const char* for wich compiler creates temporary objects